### DISCUSSION TITLE
Proper way to use next/link with flowbite-react

### DISCUSSION THREAD

**Discussion by upview** [themesberg/flowbite-react#592] [Q&A] [answered]
Hey guys,
Congratulations on the library, it's just awesome. I was wondering for the recommended way to implement next/link with flowbite-react, for components like the Navbar. Documentation [here](https://flowbite.com/docs/getting-started/next-js/) provides the following example:
`<Navbar.Link
      href="/navbars"
      active={true}
    >
      Home
    </Navbar.Link>
`
I think that's not implementing the next/link, causing the navbar to re-render while navigating the app. Can you advise me on the proper approach?
Thanks!

**Comment by lgtmrobo** [comment]
Any update here?

**Comment by tulup-conner** [comment, marked as answer]
We haven't got this feature on every component, but you can cast many components to any generic component and pass its props.

For example, see https://github.com/themesberg/flowbite-react/blob/main/src/components/Sidebar/SidebarItem.tsx#L83. You can implement this as

```js
import Link from "next/link";

<Sidebar.Item as={Link} href="/" prefetch={false}>Next.js Link</Sidebar.Item>
```

If you find a component that should have this but you can't, please do create an issue and we'll fix that up.

**Reply by VladBrok** [reply]
this is very helped me, thank you!

**Reply by idc77** [reply]
But, react-router-dom for instance the <Link> doesn't have a href attribute, but a to attribute.

**Reply by flyingduck92** [reply]
I am looking for this for days. This thing should be on the official docs

**Reply by showaltb** [reply]
I'm very much a beginner with all of this, but I was able to get this working with react-router-dom by wrapping the `<Sidebar.Item>` in a `<Link>` and casting the `<Sidebar.Item>` as a `div` to avoid browser warning about nesting `<a>` within `<a>`.

```jsx
<Link to="/dashboard">
   <Sidebar.Item as="div">Dashboard</Sidebar.Item>
</Link>
```
